modules/images/webp-uploads/load.php

```diff
@@ -105,9 +105,11 @@ function webp_uploads_create_sources_property( array $metadata, $attachment_id )
 		return $metadata;
 	}
 
+	$sizes_with_mime_type_support = webp_uploads_get_image_sizes_with_additional_mime_type_support();
+
 	foreach ( $metadata['sizes'] as $size_name => $properties ) {
```

Comment:
how about filtering the sizes you iterate through ahead of time, filtering out any where the key isn't in $sizes_with_mime_type_support?  

Q: This loop already only applies to non primary mime types, right?

Reply:
> how about filtering the sizes you iterate through ahead of time, filtering out any where the key isn't in $sizes_with_mime_type_support?

I think we don't need to do it because we return early anyway.

> Q: This loop already only applies to non primary mime types, right?

The `sizes` property contains all sizes that an image has including non primary sizes and mime types too.